cmake/tpl/FindPETSC.cmake

```diff
@@ -13,764 +11,109 @@
-  set (PETSC_INCLUDES_   ${petsc_includes_needed}  CACHE INTERNAL "PETSC include paths to be used" FORCE)
-  set (PETSC_LIBRARIES_  ${petsc_libraries_needed} CACHE INTERNAL "PETSC libraries to be used"     FORCE)
-
-  # Note that we have forced values for all these choices.  If you
-  # change these, you are telling the system to trust you that they
-  # work.  It is likely that you will end up with a broken build.
-  mark_as_advanced (PETSC_CURRENT PETSC_COMPILER PETSC_DEFINITIONS PETSC_MPIEXEC PETSC_EXECUTABLE_RUNS)
-
-endif ()
-
+include (FindPackageHandleStandardArgs)
 find_package_handle_standard_args (PETSC
-  REQUIRED_VARS PETSC_EXECUTABLE_RUNS
+  REQUIRED_VARS PETSC_FOUND PETSC_INCLUDE_DIRS PETSC_LIBRARIES
   VERSION_VAR PETSC_VERSION
-  FAIL_MESSAGE "PETSC could not be found.")
+  )
+
+if(NOT TARGET SUNDIALS::PETSC)
+  add_library(SUNDIALS::PETSC INTERFACE IMPORTED)
+  set_target_properties(SUNDIALS::PETSC PROPERTIES
+    INTERFACE_INCLUDE_DIRECTORIES "${PETSC_INCLUDE_DIRS}"
+    INTERFACE_LINK_LIBRARIES "${PETSC_LIBRARIES}"
+    )
+endif()
 
-# Create targets
-if (PETSC_FOUND)
-  if (PETSC_LIBRARY_SINGLE)
-    foreach (suffix SYS VEC MAT DM KSP SNES TS ALL)
-      if (NOT TARGET SUNDIALS::PETSC_${suffix})
-        add_library (SUNDIALS::PETSC_${suffix} UNKNOWN IMPORTED)
-        # add properties one-by-one for easier debugging
-        set_target_properties (SUNDIALS::PETSC_${suffix} PROPERTIES
-          INTERFACE_INCLUDE_DIRECTORIES "${PETSC_INCLUDES_}")
-        set_target_properties (SUNDIALS::PETSC_${suffix} PROPERTIES
-          INTERFACE_LINK_LIBRARIES "${PETSC_LIBRARIES_}")
-        set_target_properties (SUNDIALS::PETSC_${suffix} PROPERTIES
-          INTERFACE_COMPILE_OPTIONS ${PETSC_DEFINITIONS})
-        set_target_properties (SUNDIALS::PETSC_${suffix} PROPERTIES
-          IMPORTED_LOCATION ${PETSC_LIBRARY_SINGLE})
-      endif ()
-    endforeach ()
-  else ()
-    foreach (suffix SYS VEC MAT DM KSP SNES TS ALL)
-      if (PETSC_LIBRARY_${suffix} AND (NOT TARGET SUNDIALS::PETSC_${suffix}))
-        add_library (SUNDIALS::PETSC_${suffix} UNKNOWN IMPORTED)
-        # add properties one-by-one for easier debugging
-        set_target_properties (SUNDIALS::PETSC_${suffix} PROPERTIES
-          INTERFACE_INCLUDE_DIRECTORIES "${PETSC_INCLUDES_}")
-        set_target_properties (SUNDIALS::PETSC_${suffix} PROPERTIES
-          INTERFACE_LINK_LIBRARIES "${PETSC_LIBRARIES_${suffix}}")
-        set_target_properties (SUNDIALS::PETSC_${suffix} PROPERTIES
-          INTERFACE_COMPILE_OPTIONS ${PETSC_DEFINITIONS})
-        set_target_properties (SUNDIALS::PETSC_${suffix} PROPERTIES
-          IMPORTED_LOCATION ${PETSC_LIBRARY_${suffix}})
-      endif ()
-    endforeach ()
-  endif ()
-endif (PETSC_FOUND)
+mark_as_advanced(PETSC_INCLUDE_DIRS PETSC_LIBRARIES PETSC_VERSION_MAJOR PETSC_VERSION_MINOR PETSC_VERSION_PATCH VERSION_VAR PETSC_VERSION)
```

Comment:
Is `VERSION_VAR` needed? I didn't see it above.

Reply:
Nope, not needed anymore.